### DISCUSSION TITLE
Don't start RemoteAPP - RAIL exec error: execResult=RAIL_EXEC_E_DECODE_FAILED NtError=0xD

### DISCUSSION THREAD

**Discussion by Yaguar666** [FreeRDP/FreeRDP#6776] [Q&A]
Trying connect to RemoteAPP on Windows Server 2012 R2.
From Windows it works, but from ubuntu - doesn't.

Client Version:
FreeRDP version 2.2.0 (n/a) and FreeRDP version 1.1.0-beta1 (git n/a)

Command:
xfreerdp /u:user /p:password /d:terminal.ru /v:test.terminal.ru:3389 /cert-ignore /load-balance-info:"tsv://MS Terminal Services Plugin.1.Test" /bpp:32 /f /multitouch +compression +fonts /app:||1cestart  /app-cmd:'ENTERPRISE /S"DEV\V8_DEV" /N"cash" /P"Cash"  /RunModeOrdinaryApplication'

ERROR:
[16:50:03:034] [5545:5558] [ERROR][com.freerdp.client.x11] - RAIL exec error: execResult=RAIL_EXEC_E_DECODE_FAILED NtError=0xD
[16:50:03:034] [5545:5546] [ERROR][com.freerdp.core] - freerdp_check_fds() failed - 0
[16:50:03:034] [5545:5546] [INFO][com.freerdp.client.common] - Network disconnect!
[16:50:03:034] [5545:5546] [ERROR][com.freerdp.client.x11] - Failed to check FreeRDP file descriptor
1cestart: command not found

OR:
RAIL exec error: execResult=RAIL_EXEC_E_DECODE_FAILED NtError=0xD

RemoteAPP server configuration:
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Terminal Server\TSAppAllowList]
"fDisableAllowList"=dword:00000000

HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Terminal Server\TSAppAllowList\Application\1cestart]
"CommandLineSetting"=dword:00000002
"RequiredCommandLine"="ENTERPRISE /S"DEV\V8_DEV" /N"cash" /P"Cash"  /RunModeOrdinaryApplication"
"Path"="C:\\Program Files (x86)\\1cv8\\common\\1cestart.exe"
"ShortPath"="C:\\PROGRA~2\\1cv8\\common\\1cestart.exe"
"ShowInTSWA"=dword:00000000
"Name"="1C V8_DEV"
"SecurityDescriptor"=""
"VPath"="%SYSTEMDRIVE%\\Program Files (x86)\\1cv8\\common\\1cestart.exe"

Google didn't help(

**Comment by akallabeth** [comment]
Looks like you use the wrong start command? It does not match the one mentioned in your allow list

Am 2. Februar 2021 15:02:45 UTC schrieb Egor Savinov <notifications@github.com>:

**Reply by Yaguar666** [reply]
Can you specify which parameter is wrong?
Application name is same, application start command is same...

**Reply by Yaguar666** [reply]
And I still get an error, if I make appllication without start command.

**Reply by akallabeth** [reply]
you have full paths in the server settings, but try relative ones when starting.
they both have to match exactly in order to work, otherwise the server 'does not know' which command you are trying to execute

**Reply by Yaguar666** [reply]
If I connect with full paths - 
`xfreerdp /u:user /p:password /d:terminal.ru /v:test.terminal.ru:3389 /cert-ignore /load-balance-info:"tsv://MS Terminal Services Plugin.1.Test" /bpp:32 +fonts /app:||'C:\Program Files (x86)\1cv8\common\1cestart.exe' /app-cmd:'ENTERPRISE /S"DEV\V8_DEV" /N"cash" /P"Cash" /RunModeOrdinaryApplication'`
I get an error - 
`RAIL exec error: execResult=RAIL_EXEC_E_NOT_IN_ALLOWLIST NtError=0x15`
Because application isn't in allowed list.

My Windows rdp link is (work perfectly):
```
session bpp:i:32
remoteapplicationmode:i:1
server port:i:3389
allow font smoothing:i:1
full address:s:test.terminal.ru
remoteapplicationprogram:s:||1cestart
remoteapplicationname:s:1C V8_DEV
remoteapplicationcmdline:s:ENTERPRISE /S"DEV\V8_DEV" /N"cash" /P"Cash"  /RunModeOrdinaryApplication
loadbalanceinfo:s:tsv://MS Terminal Services Plugin.1.Test
```
And link with fullpaths (also work perfectly):
```
session bpp:i:32
remoteapplicationmode:i:1
server port:i:3389
allow font smoothing:i:1
full address:s:test.terminal.ru
remoteapplicationprogram:s:C:\Program Files (x86)\1cv8\common\1cestart.exe
remoteapplicationname:s:1C V8_DEV
remoteapplicationcmdline:s:ENTERPRISE /S"DEV\V8_DEV" /N"cash" /P"Cash"  /RunModeOrdinaryApplication 
loadbalanceinfo:s:tsv://MS Terminal Services Plugin.1.Test
```
After reading documentation and forums - I see two ways:
1. Add application in allowed list & connect at alias - not work in freerdp
2. Block allowed list (allow all application) & connect at full paths name - work in freerdp but it's not safe(

**Reply by Yaguar666** [reply]
Maybe connect RemoteAPP with alias is broken?

**Reply by akallabeth** [reply]
Strange, this is the way we usually configure our servers for testing. (The allow list with alias)
Client side there are not that many options (mostly just forward the strings to the server).
Did you try using the `.rdp` file directly?
Also, could you try the nightly builds, just be sure that it is not a bug already fixed you are running into?

**Reply by Yaguar666** [reply]
Nightly build get same rail plugin error.
But if i use `.rdp` file directly, 2.0.0, 2.2.0 and nightly versions work fine (nightly work very fast like on windows :) )
I think using `.rdp` file directly is the solution to my problem.
Ty for help)
